edb/edgeql/compiler/viewgen.py

```diff
@@ -244,6 +250,100 @@ def _process_view(
             )
         )
 
+    explicit_ptr_names = {
+        desc.ptr_name for desc in shape_desc if not desc.is_linkprop
+    }
+
+    explicit_lprop_names = {
+        desc.ptr_name for desc in shape_desc if desc.is_linkprop
+    }
+
+    # Now look for any splats and expand them.
```

Comment:
Do you think it would be cleaner to break some of this off into a function?
(Maybe not)